extensions/amp-date-picker/1.0/date-range-picker.js

```diff
@@ -0,0 +1,263 @@
+import {addDays, differenceInDays, isAfter, isSameDay} from 'date-fns';
+
+import {
+  closestAncestorElementBySelector,
+  scopedQuerySelector,
+} from '#core/dom/query';
+
+import * as Preact from '#preact';
+import {useCallback, useEffect, useMemo, useRef, useState} from '#preact';
+import {Children} from '#preact/compat';
+import {ContainWrapper} from '#preact/component';
+
+import {BaseDatePicker} from './base-date-picker';
+import {
+  DateFieldNameByType,
+  DateFieldType,
+  DatePickerMode,
+  FORM_INPUT_SELECTOR,
+  TAG,
+} from './constants';
+import {getFormattedDate, parseDate} from './date-helpers';
+
+/**
+ * @param {!DateInput.Props} props
+ * @return {PreactDef.Renderable}
+ */
+export function DateRangePicker({
+  allowBlockedEndDate,
+  allowBlockedRanges,
+  blockedDates,
+  children,
+  endInputSelector,
+  format,
+  id,
+  mode,
+  onError,
+  startInputSelector,
+  ...rest
+}) {
+  const [startDate, setStartDate] = useState();
+  const [endDate, setEndDate] = useState();
+
+  const [startInputAttributes, setStartInputAttributes] = useState();
+  const [endInputAttributes, setEndInputAttributes] = useState();
+
+  const onErrorRef = useRef(onError);
+  const containerRef = useRef();
+
+  const [isOpen] = useState(mode === DatePickerMode.STATIC);
```

Comment:
i assume this is just a placeholder for later.

Reply:
Yes, I haven't pulled in the rest of the state logic yet.